### DISCUSSION TITLE
Routing matching

### DISCUSSION THREAD

**Discussion by MataSong** [fastapi/fastapi#9637] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
`urlpatterns = [
    re_path(r'^(?:.*)/?$', views.index_view, name='index')
]`
```

### Description

Is there a method for FASTAPI to match all frontend routes using regularization like django? 

### Operating System

Windows

### FastAPI Version

0.96.0

### Python Version

python 3.10

**Comment by chrisK824** [comment]
Hey @MataSong , not sure what you are after here, but you can group your routes using api router feature and even add a specific prefix to each router, you can see the example in the documentation https://fastapi.tiangolo.com/tutorial/bigger-applications/

**Comment by esrefzeki** [comment]
If I understand it correctly:

In FastAPI, you can achieve a similar behavior to Django's regular expression-based URL routing by using a catch-all route. However, unlike Django's regular expression-based routing, FastAPI uses path parameters rather than regular expressions for routing.

Here's an example of how you can define a catch-all route in FastAPI:

```
from fastapi import FastAPI

app = FastAPI()

@app.get("/{path:path}")
async def catch_all(path: str):
    # Your logic for handling the catch-all route
    return {"path": path}
```

In this example, the route '/{path:path}' defines a path parameter named path with the type 'str'. The ':path' suffix in the parameter declaration is used to specify that the parameter should match any path, including slashes ('/'). By doing this, any incoming request path will be matched by this route, allowing you to handle all frontend routes.

The catch-all route handler, 'catch_all', will receive the matched path as the path parameter. You can then implement the necessary logic to handle the catch-all route, such as returning a response or redirecting to a different page.

It's important to note that this catch-all route should be defined as the last route in your application, as it will match all paths and override any subsequent routes defined in your FastAPI application.

**Reply by MataSong** [reply]
I tried using your method, but when I return to the page, an error will be reported.

`
@app.get("/{path:path}")
async def catch_all(request: Request):
    return HTMLResponse(
        path("html_path").read_text(),
        # media_type="text/html",
        status_code=status.HTTP_200_OK
    )
`

error:   runtime.796d2a9187f79004.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
polyfills.2d63083413045318.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
main.6da756f47e826a20.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.
scripts.8d06af132615268d.js:1 Uncaught SyntaxError: Unexpected token '<' (at scripts.8d06af132615268d.js:1:1)

**Reply by 2h311** [reply]
What are you trying to do?

**Reply by MataSong** [reply]
> 你想做什么？

Currently, I have an Angular front-end project that I have built into a static file and placed in the backend project of FastAPI. I need the backend to be able to use and match the routing to the front-end.

**Reply by jgould22** [reply]
You should use static files for that.

https://fastapi.tiangolo.com/tutorial/static-files/

**Comment by jgould22** [comment]
Starlette/FastAPI does not have this functionality at this time. 

https://github.com/encode/starlette/issues/1101

**Comment by 2h311** [comment, marked as answer]
Just tried doing something like this in FastAPI now since I've done it in Django before; it turns out it is not available in FastAPI at the moment.
